src/llmq/quorums.cpp

```diff
@@ -211,13 +221,15 @@ void CQuorumManager::TriggerQuorumDataRecoveryThreads(const CBlockIndex* pIndex)
             }
 
             uint16_t nDataMask{0};
-            const bool fWeAreQuorumMember = pQuorum->IsValidMember(activeMasternodeInfo.proTxHash);
+            const bool fWeAreQuorumMember = WITH_LOCK(activeMasternodeInfoCs, return pQuorum->IsValidMember(activeMasternodeInfo.proTxHash));
             const bool fSyncForTypeEnabled = mapQuorumVvecSync.count(pQuorum->qc->llmqType) > 0;
             const QvvecSyncMode syncMode = fSyncForTypeEnabled ? mapQuorumVvecSync.at(pQuorum->qc->llmqType) : QvvecSyncMode::Invalid;
             const bool fSyncCurrent = syncMode == QvvecSyncMode::Always || (syncMode == QvvecSyncMode::OnlyIfTypeMember && fWeAreQuorumTypeMember);
 
-            if ((fWeAreQuorumMember || (fSyncForTypeEnabled && fSyncCurrent)) && pQuorum->quorumVvec == nullptr) {
-                nDataMask |= llmq::CQuorumDataRequest::QUORUM_VERIFICATION_VECTOR;
+            {
+                if ((fWeAreQuorumMember || (fSyncForTypeEnabled && fSyncCurrent)) && WITH_LOCK(pQuorum->quorumVvecCs, return pQuorum->quorumVvec == nullptr)) {
```

Comment:
No need for redundant scope if we use WITH_LOCK right?

Reply:
Right.